cli/src/main.rs

```diff
@@ -112,25 +46,43 @@ pub async fn serve(opts: Opts) -> Result<(), Error> {
 }
 
 #[tokio::main]
-pub async fn main() -> Result<(), Error> {
+pub async fn main() -> ExitCode {
     // Parse the command-line options, exiting if there are any errors
     let opts = Opts::parse();
-
     install_tracing_subscriber(&opts);
-
-    tokio::select! {
-        _ = tokio::signal::ctrl_c() => {
-            Ok(())
+    if opts.run_mode() {
+        // println!("Using Viceroy to run tests...");
+        match run_wasm_main(opts).await {
+            Ok(_) => ExitCode::SUCCESS,
+            Err(_) => ExitCode::FAILURE,
```

Comment:
It'd be nice to translate the WASI exit code to the native exit code if possible. It looks like that'd be a fair amount of plumbing and potentially some downcasting from dynamic error types, so I'd say it can wait for a future PR if you note it as an issue so we don't forget.

Reply:
Follow-up issue: https://github.com/fastly/Viceroy/issues/223

Reply:
Thanks for capturing that!